sphinx_gallery/utils.py

```diff
@@ -36,3 +37,16 @@ def __init__(self):  # noqa: D102
 
     def __del__(self):  # noqa: D105
         rmtree(self._path, ignore_errors=True)
+
+
+def replace_py_ipynb(fname):
```

Comment:
Isn't it just simpler to have a regex substitution
```python
re.sub('.py$', '.ipynb', fname)
```
We have already filtered for python files when collecting files for examples. No need for the value error check.

Reply:
I like the regex, implemented in commit below.  But a little extra early-checking seems safe.  If you want me to rip it out, I can do that too.